From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Sat, 18 Jun 2016 19:27:49 -0400
Subject: ipsec newhostkey: default to /etc/ipsec.d/newhostkey-generated.secret

By default, avoid touching /etc/ipsec.secrets.  Instead, generate a
.secret file that will be included by the default /etc/ipsec.secrets.

Also, document the change in defaults
---
 programs/newhostkey/ipsec_newhostkey.8.xml | 5 +++--
 programs/newhostkey/newhostkey.in          | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/programs/newhostkey/ipsec_newhostkey.8.xml b/programs/newhostkey/ipsec_newhostkey.8.xml
index b24bd70..87ac766 100644
--- a/programs/newhostkey/ipsec_newhostkey.8.xml
+++ b/programs/newhostkey/ipsec_newhostkey.8.xml
@@ -63,8 +63,9 @@
 	  exist, it is created under umask <emphasis
 	  remap='B'>077</emphasis>.  If the file already exists and is
 	  non-empty, a warning message about that is written to
-	  standard error, and the output is appended to the
-	  file.</para>
+	  standard error, and the output is appended to the file.  The
+	  default is <emphasis
+	  remap='I'>/etc/ipsec.d/newhostkey-generated.secrets</emphasis>.</para>
 	</listitem>
       </varlistentry>
 
diff --git a/programs/newhostkey/newhostkey.in b/programs/newhostkey/newhostkey.in
index 8e867ef..2fcb358 100755
--- a/programs/newhostkey/newhostkey.in
+++ b/programs/newhostkey/newhostkey.in
@@ -26,7 +26,7 @@ bits=
 verbose=
 host=
 seeddev="--seeddev /dev/random"
-output=""
+output="@FINALCONFDIR@/ipsec.d/newhostkey-generated.secrets"
 configdir="@FINALNSSDIR@"
 password=
 for dummy; do
